HyperVExtension/src/HyperVExtension/Models/VMGalleryCreationAdaptiveCardSession.cs

```diff
@@ -0,0 +1,341 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Runtime.InteropServices.WindowsRuntime;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using HyperVExtension.Common;
+using HyperVExtension.Exceptions;
+using HyperVExtension.Helpers;
+using HyperVExtension.Models.VirtualMachineCreation;
+using HyperVExtension.Models.VMGalleryJsonToClasses;
+using Microsoft.Windows.DevHome.SDK;
+using Serilog;
+using Windows.Foundation;
+using Windows.Storage;
+using Windows.Storage.Streams;
+
+namespace HyperVExtension.Models;
+
+public enum SessionState
+{
+    InitialCreationForm,
+    ReviewForm,
+}
+
+public class VMGalleryCreationAdaptiveCardSession : IExtensionAdaptiveCardSession2
+{
+    private readonly Serilog.ILogger _log = Log.ForContext("SourceContext", nameof(HyperVVirtualMachine));
+
+    private readonly string _pathToInitialCreationFormTemplate = Path.Combine(AppContext.BaseDirectory, @"HyperVExtension\Templates\", "InitialVMGalleryCreationForm.json");
+
+    private readonly string _pathToReviewFormTemplate = Path.Combine(AppContext.BaseDirectory, @"HyperVExtension\Templates\", "ReviewFormForVMGallery.json");
+
+    private readonly string _adaptiveCardNextButtonId = "DevHomeMachineConfigurationNextButton";
+
+    /// <summary>
+    /// The gallery images that will be displayed in the initial creation form. We retrieve these from the VMGallery.json file in the Microsoft servers.
+    /// </summary>
+    private readonly VMGalleryImageList _vMGalleryImageList;
+
+    private readonly IStringResource _stringResource;
+
+    /// <summary>
+    /// Gets the Json string that represents the user input that was passed to the adaptive card session. We'll keep this so we can pass it back to Dev Home
+    /// at the end of the session.
+    /// </summary>
+    public string OriginalUserInputJson { get; private set; } = string.Empty;
+
+    public event TypedEventHandler<IExtensionAdaptiveCardSession2, ExtensionAdaptiveCardSessionStoppedEventArgs>? Stopped;
+
+    public void Dispose()
+    {
+    }
+
+    public VMGalleryCreationAdaptiveCardSession(VMGalleryImageList galleryImages, IStringResource stringResource)
+    {
+        _vMGalleryImageList = galleryImages;
+        _stringResource = stringResource;
+    }
+
+    private IExtensionAdaptiveCard? _creationAdaptiveCard;
+
+    public bool ShouldEndSession { get; private set; }
+
+    public ProviderOperationResult Initialize(IExtensionAdaptiveCard extensionUI)
+    {
+        _creationAdaptiveCard = extensionUI;
+
+        return GetInitialCreationFormAdaptiveCard();
+    }
+
+    public IAsyncOperation<ProviderOperationResult> OnAction(string action, string inputs)
+    {
+        return Task.Run(async () =>
+        {
+            ProviderOperationResult operationResult;
```

Comment:
nit; Initialize variables when they are needed. adaptiveCardStateNotRecognizedError is only used in ProviderOperationResult, could be inlined there.